app/controllers/users_controller.rb

```diff
@@ -1,5 +1,6 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show liked feed followers following discover ]
+  before_action :ensure_current_user_is_owner, only: [:feed, :discover]
```

Comment:
haven't seen people do the UsersController yet but i am liking this implementation here!